Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/CapsuleViewportEdit.cpp

```diff
@@ -24,68 +24,194 @@ namespace AzToolsFramework
+        SetupHeightManipulator(manipulatorManagerId, worldTransform, localTransform, nonUniformScale);
+    }
+
+    void CapsuleViewportEdit::AddEntityComponentIdPair(const AZ::EntityComponentIdPair& entityComponentIdPair)
+    {
+        if (m_heightManipulator)
+        {
+            m_heightManipulator->AddEntityComponentIdPair(entityComponentIdPair);
+        }
+        if (m_radiusManipulator)
+        {
+            m_radiusManipulator->AddEntityComponentIdPair(entityComponentIdPair);
+        }
+        AZ_WarningOnce(
+            "CapsuleViewportEdit",
+            m_heightManipulator && m_radiusManipulator,
+            "Attempting to AddEntityComponentIdPair before manipulators have been created");
     }
 
-    void CapsuleViewportEdit::UpdateCapsuleManipulators()
+    void CapsuleViewportEdit::UpdateManipulators()
     {
-        AZ::Transform worldTransform = GetCapsuleWorldTransform();
-        AZ::Vector3 nonUniformScale = GetCapsuleNonUniformScale();
-        AZ::Transform localTransform = GetCapsuleLocalTransform();
+        const AZ::Transform worldTransform = GetManipulatorSpace();
+        const AZ::Vector3 nonUniformScale = GetNonUniformScale();
+        const AZ::Transform localTransform = GetLocalTransform();
 
         const float capsuleHeight = GetCapsuleHeight();
         const float capsuleRadius = GetCapsuleRadius();
 
-        m_radiusManipulator->SetSpace(worldTransform);
-        m_radiusManipulator->SetLocalTransform(
-            localTransform * AZ::Transform::CreateTranslation(m_radiusManipulator->GetAxis() * capsuleRadius));
-        m_radiusManipulator->SetNonUniformScale(nonUniformScale);
-        m_heightManipulator->SetSpace(worldTransform);
-        m_heightManipulator->SetLocalTransform(
-            localTransform * AZ::Transform::CreateTranslation(m_heightManipulator->GetAxis() * capsuleHeight * HalfHeight));
-        m_heightManipulator->SetNonUniformScale(nonUniformScale);
+        if (m_radiusManipulator)
+        {
+            m_radiusManipulator->SetSpace(worldTransform);
+            m_radiusManipulator->SetLocalTransform(
+                localTransform * AZ::Transform::CreateTranslation(m_radiusManipulator->GetAxis() * capsuleRadius));
+            m_radiusManipulator->SetNonUniformScale(nonUniformScale);
+        }
+        if (m_heightManipulator)
+        {
+            m_heightManipulator->SetSpace(worldTransform);
+            m_heightManipulator->SetLocalTransform(
+                localTransform * AZ::Transform::CreateTranslation(m_heightManipulator->GetAxis() * capsuleHeight * HalfHeight));
+            m_heightManipulator->SetNonUniformScale(nonUniformScale);
+        }
     }
 
-    void CapsuleViewportEdit::ResetCapsuleManipulators()
+    void CapsuleViewportEdit::ResetValues()
     {
+        BeginEditing();
         SetCapsuleHeight(ResetCapsuleHeight);
         SetCapsuleRadius(ResetCapsuleRadius);
+        FinishEditing();
```

Comment:
question: I take it this is to handle a potential undo action?

Reply:
Yeah it's so you can undo reset, although I've just realised this won't be connected to the undo/redo system in the main editor. I fixed a similar issue for box component mode the other day, but thinking about it again now it's probably better to fix it at the lower level. Anyway, I'm about to touch this code again and I have a passing AR, so I'll address that in the next PR.